src/main/java/edu/umich/its/cpm/MigrationController.java

```diff
@@ -302,6 +308,57 @@ private String get_user_myworkspace_site_json(HttpServletRequest request) {
 	}
 
 	/**
+	 * retain only the site record where the current user has Owner role inside
+	 * @param request
+	 * @param projectsString
+	 * @param currentUserId
+	 * @return
+	 */
+	private String filterForSitesWithOwnerRole(HttpServletRequest request,
+			String projectsString, String currentUserId) {
+		JSONArray ownerSitesJSONArray = new JSONArray();
+		try {
+			JSONObject sitesJSONObject = new JSONObject(projectsString);
+			// get site array
+			JSONArray sitesJSONArray = sitesJSONObject.getJSONArray("site_collection");
+			
+			// filter out those sites that current user has role "Owner" in it
+			for (int iSite = 0; sitesJSONArray != null && iSite < sitesJSONArray.length(); ++iSite) {
+				JSONObject siteJSON = sitesJSONArray.getJSONObject(iSite);
+				String siteId = siteJSON.getString("id");
+				try
+				{
+					// get all site members
+					HashMap<String, String> userRoles = get_site_members(
+							request, siteId);
+					for (String userEid : userRoles.keySet()) {
+						String userRole = userRoles.get(userEid);
+						if (Utils.ROLE_OWNER.equals(userRole) && userEid.equals(currentUserId))
+						{
+							// keep the site JSON if current user has Owner role in this site
+							ownerSitesJSONArray.put(siteJSON);
+							break;
+						}
+					}
+				}
+				catch (Exception e)
```

Comment:
Would it be more beneficial to specify the type of exception here rather than just having the generic exception being caught?